### DISCUSSION TITLE
Glowroot giving 502 error -unable to connect to central collector

### DISCUSSION THREAD

**Discussion by adeokar13** [glowroot/glowroot#941] [Q&A] [answered]
Hi,

i am getting 502 error while connecting to glowroot when checked glowroot logs found below error

2022-06-01 08:47:12.099 WARN  o.g.agent.central.CentralConnection - unable to establish connection with the central collector IP address :8181 (will keep trying...): org.glowroot.agent.shaded.io.netty.channel.ConnectTimeoutException: connection timed out: /IP address :8181

can someone please help?

**Comment by Never-lab** [comment, marked as answer]
The agent log is the real signal: `ConnectTimeoutException` to `…:8181`. That is **network / wrong host / firewall / Central not listening on gRPC**, not a UI 502 bug by itself.

Checklist:
1. `collector.address` = Central **gRPC** port (default **8181**), not the UI port (**4000**).
2. From the app host: TCP reachability to that host:port.
3. Same major line agent ↔ Central; prefer [0.14.7](https://github.com/glowroot/glowroot/releases/tag/v0.14.7).

Wiki: [Agent Installation (for Central Collector)](https://github.com/glowroot/glowroot/wiki/Agent-Installation-(for-Central-Collector)).
